src/Blueprint/Blueprint/Models/PSBlueprint.cs

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.Blueprint.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Common;
+using System.Collections;
+
+namespace Microsoft.Azure.Commands.Blueprint.Models
+{
+    public class PSBlueprint : PSBlueprintBase
+    {
+        public string DefinitionLocationId { get; set; }
+        public string Scope { get; set; }
+        public object Versions { get; set; }
+        public List<string> ParametersDisplayList { get; set; }
+        public List<string> ResourceGroupDisplayList { get; set; }
+        public List<string> VersionList { get; set; }
+
+        /// <summary>
+        /// Create a PSBlueprint object from a BlueprintModel.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <param name="managementGroupName">Name of the management group the blueprint belongs to.</param>
+        /// <returns>A new PSBlueprint object</returns>
+        internal static PSBlueprint FromBlueprintModel(BlueprintModel model, string scope)
+        {
+            var psBlueprint = new PSBlueprint
+            {
+                Id = model.Id,
+                Name = model.Name,
+                DefinitionLocationId = Utils.GetDefinitionLocationId(scope),
+                Scope = scope,
+                DisplayName = model.DisplayName,
+                Description = model.Description,
+                Status = new PSBlueprintStatus(),
+                TargetScope = PSBlueprintTargetScope.Unknown,
+                Parameters = new Dictionary<string, PSParameterDefinition>(),
+                ResourceGroups = new Dictionary<string, PSResourceGroupDefinition>(),
+                Versions = model.Versions,
+                ParametersDisplayList = new List<string>(),
+                ResourceGroupDisplayList = new List<string>(),
+                VersionList = new List<string>()
+            };
+
+            if (DateTime.TryParse(model.Status.TimeCreated, out DateTime timeCreated))
+            {
+                psBlueprint.Status.TimeCreated = timeCreated;
+            }
+            else
+            {
+                psBlueprint.Status.TimeCreated = null;
+            }
+
+            if (DateTime.TryParse(model.Status.LastModified, out DateTime lastModified))
+            {
+                psBlueprint.Status.LastModified = lastModified;
+            }
+            else
+            {
+                psBlueprint.Status.LastModified = null;
+            }
+
+            if (Enum.TryParse(model.TargetScope, true, out PSBlueprintTargetScope targetScope))
+            {
+                psBlueprint.TargetScope = targetScope;
+            }
+            else
+            {
+                psBlueprint.TargetScope = PSBlueprintTargetScope.Unknown;
```

Comment:
>Unknown [](start = 65, length = 7)

This is fine for now. But in which scenario will any of the else branches in this function actually be taken?

Reply:
Resolving for now. I'll follow up on this offline.

---
In reply to: [258257485](https://github.com/Azure/azure-powershell/pull/8537#discussion_r258257485) [](ancestors = 258257485)